src/test/java/com/google/devtools/build/lib/rules/cpp/SkylarkCcCommonTest.java

```diff
@@ -5734,23 +5703,16 @@ private static void createFiles(
         ")");
   }
 
-  private void doTestWrongExtensionOfSrcsAndHdrs(String attrName) throws Exception {
-    createFiles(scratch, "tools/build_defs/foo");
-    scratch.file(
-        "bar/BUILD",
-        "load('//tools/build_defs/foo:extension.bzl', 'cc_skylark_library')",
-        "cc_skylark_library(",
-        "    name = 'skylark_lib',",
-        "    " + attrName + " = ['skylark_lib.cannotpossiblybevalid'],",
-        ")");
-    reporter.removeHandler(failFastHandler);
-    getConfiguredTarget("//bar:skylark_lib");
-    assertContainsEvent(
-        "has wrong extension. The list of possible extensions for '" + attrName + "'");
-  }
-
-  private void doTestPossibleExtensionsOfSrcsAndHdrs(String attrName, List<String> extensions)
+  private void doTestPossibleExtensionsOfSrcsAndHdrs(String attrName, List<String> knownExtensions)
       throws Exception {
+    List<String> extensions = new ArrayList<String>(knownExtensions);
+    String unknownExtension = ".unknown_extension";
```

Comment:
Test passed, but bazel not working with unknown header extensions with following stack trace: 

```
Caused by: java.lang.IllegalStateException
        at com.google.common.base.Preconditions.checkState(Preconditions.java:491)
        at com.google.devtools.build.lib.rules.cpp.CcCompilationHelper.addPrivateHeader(CcCompilationHelper.java:424)
        at com.google.devtools.build.lib.rules.cpp.CcCompilationHelper.addPrivateHeaders(CcCompilationHelper.java:408)
        at com.google.devtools.build.lib.rules.cpp.CcModule.compile(CcModule.java:1556)
        at com.google.devtools.build.lib.bazel.rules.cpp.BazelCcModule.compile(BazelCcModule.java:88)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
```

It's because failing precondition in `com.google.devtools.build.lib.rules.cpp.CcCompilationHelper.addPrivateHeader`:
```
    Preconditions.checkState(isHeader || isTextualInclude);
```

Looks like this test is not correct and it test only first element in `extensions` list. If I add `invalidatePackages();` in loop below, test will fail (same as bazel binary).

Could someone say is it correct way of fixing the test? Or maybe I didn't understand something.